app/routes/staff/products/index.tsx

```diff
@@ -1,6 +1,198 @@
+  "フルーツ",
+  "炭酸",
+  "デザート",
+  "冷たい",
+]
+
+const totalProducts = products.length
+const lowStockCount = products.filter((p) => p.stock <= 5 && p.stock > 0).length
+const outOfStockCount = products.filter((p) => p.stock === 0).length
+const totalValue = products.reduce((sum, p) => sum + p.price * p.stock, 0)
+
+const productFormDataToProduct = (formData: FormData): Omit<Product, "id"> => {
+  const name = String(formData.get("name") ?? "").trim()
+  if (!name) throw new Error("商品名は必須です")
+  if (name.length > 100)
+    throw new Error("商品名は100文字以内で入力してください")
+
+  const image = String(formData.get("image") ?? "").trim()
+  if (image && !/^https?:\/\/.+\.(jpg|jpeg|png|gif|webp)$/i.test(image)) {
+    throw new Error(
+      "画像URLが不正です（http(s)で始まり画像拡張子で終わる必要があります）",
+    )
+  }
+
+  const price = Number(formData.get("price"))
+  if (Number.isNaN(price) || price < 0 || !Number.isInteger(price)) {
+    throw new Error("価格は0以上の整数で入力してください")
+  }
+
+  const stock = Number(formData.get("stock"))
+  if (Number.isNaN(stock) || stock < 0 || !Number.isInteger(stock)) {
+    throw new Error("在庫数は0以上の整数で入力してください")
+  }
+
+  const tags = (formData.getAll("tags") as string[])
+    .map((t) => t.trim())
+    .filter(Boolean)
+  return {
+    name,
+    image,
+    tags,
+    price,
+    stock,
+  }
+}
+
+export const POST = createRoute(async (c) => {
+  try {
+    const formData = await c.req.formData()
+    // @ts-expect-error
+    const product = productFormDataToProduct(formData)
+    // TODO: Implement product registration logic
+
+    setToastCookie(c, "success", "商品を登録しました")
+  } catch (e) {
+    setToastCookie(c, "error", String(e))
+  }
+  return c.redirect(c.req.url)
+})
 
 export default createRoute((c) => {
-  return c.render(<Page pathname={c.req.path} />)
+  const url = new URL(c.req.url)
+  const viewMode = c.req.query("view") === "card" ? "card" : "table"
+  const search = url.search
+  const { toastType, toastMessage } = getToastCookie(c)
+
+  if (toastType || toastMessage) {
+    deleteToastCookie(c)
+  }
+
+  return c.render(
+    <Layout title={"商品管理"} description={"商品情報の登録や編集を行います。"}>
+      {toastMessage && (
+        <div className="fixed top-4 right-4 z-50">
+          <Toast message={toastMessage} type={toastType} />
+        </div>
+      )}
+      <div className="min-h-screen bg-gray-50 p-4">
+        <div className="max-w-7xl mx-auto space-y-6">
+          <ProductInfo
+            totalProducts={totalProducts}
+            outOfStockCount={outOfStockCount}
+            lowStockCount={lowStockCount}
+            totalValue={totalValue}
+          />
+          <ProductRegister tags={tags} />
+          <div className="bg-white rounded-lg border p-4">
+            <div className="flex flex-col sm:flex-row items-center gap-4 mb-4">
+              <span className="text-lg font-bold flex-1">商品一覧</span>
+              <div className="flex-1 flex justify-end">
+                <ViewModeToggle viewMode={viewMode} search={search} />
```

Comment:
タブレットレイアウトで不必要な改行が発生していたので、次のように修正した方が良いと思われます

BEFORE:

<img width="1281" height="1843" alt="image" src="https://github.com/user-attachments/assets/4f6ced97-7f15-4fbe-96bb-be61629f5eda" />

AFTER:

<img width="1281" height="1843" alt="image" src="https://github.com/user-attachments/assets/1905a26b-3c93-48fb-b5fd-0421248646f5" />

```suggestion
            <div className="flex flex-col sm:flex-row items-center justify-between gap-4 mb-4">
              <span className="text-lg font-bold">商品一覧</span>
              <div className="flex justify-end">
                <ViewModeToggle viewMode={viewMode} search={search} />
```